src/components/InventoryV2/BatchDetailsModal.tsx

```diff
@@ -0,0 +1,396 @@
+              };
+            })
+          : [];
+
+        setPhotos(photosWithUrl);
+      }
+    };
+
+    getPhotos();
+  }, [batch, snackbar]);
+
+  useEffect(() => {
+    if (record) {
+      const notReadyQuantity = record?.notReadyQuantity ?? 0;
+      const readyQuantity = record?.readyQuantity ?? 0;
+      setTotalQuantity(+notReadyQuantity + +readyQuantity);
+    }
+  }, [record, selectedOrganization]);
+
+  useEffect(() => {
+    if (record?.facilityId) {
+      const newFacility = getNurseryById(selectedOrganization, record.facilityId);
+      if (newFacility.id.toString() !== facility?.id.toString()) {
+        setFacility(newFacility);
+      }
+    }
+  }, [record?.facilityId, selectedOrganization, facility?.id]);
+
+  useEffect(() => {
+    if (timeZone !== tz.id) {
+      setTimeZone(tz.id);
+    }
+  }, [tz.id, timeZone]);
+
+  useEffect(() => {
+    const foundFacility = selectedOrganization.facilities?.find(
+      (f) => f.id.toString() === batch?.facilityId.toString()
+    );
+    if (foundFacility) {
+      setFacility(foundFacility);
+    }
+  }, [batch, setRecord, selectedOrganization]);
+
+  const MANDATORY_FIELDS = ['germinatingQuantity', 'notReadyQuantity', 'readyQuantity', 'addedDate'] as const;
+  type MandatoryField = typeof MANDATORY_FIELDS[number];
+
+  const hasErrors = () => {
+    if (record) {
+      const missingRequiredField = MANDATORY_FIELDS.some((field: MandatoryField) => !record[field]);
+      return missingRequiredField;
+    }
+    return true;
+  };
+
+  const updatePhotos = async () => {
+    await NurseryBatchService.uploadBatchPhotos(batch.id, newPhotos);
+    setNewPhotos([]);
+    if (photoIdsToRemove) {
+      await NurseryBatchService.deleteBatchPhotos(batch.id, photoIdsToRemove);
+      setPhotoIdsToRemove([]);
+    }
+  };
+
+  const saveBatch = async () => {
+    if (record) {
+      if (hasErrors()) {
+        setValidateFields(true);
+        return;
+      }
+
+      let response;
+      let responseQuantities = { requestSucceeded: true };
+
+      response = await NurseryBatchService.updateBatch(record);
+      if (response.batch) {
+        responseQuantities = await NurseryBatchService.updateBatchQuantities({
+          ...record,
+          version: response.batch.version,
+        });
+      }
+
+      if (response.requestSucceeded && responseQuantities.requestSucceeded) {
+        await updatePhotos();
+        reload();
+        onCloseHandler();
+      } else {
+        snackbar.toastError();
+      }
+    }
+  };
+
+  const onCloseHandler = () => {
+    setValidateFields(false);
+    onClose();
+  };
+
+  const gridSize = () => (isMobile ? 12 : 6);
+
+  const paddingSeparator = () => (isMobile ? 0 : 1.5);
+
+  const changeDate = (id: string, value?: any) => {
+    const date = value ? getDateDisplayValue(value.getTime(), tz.id) : null;
+    onChange(id, date);
+  };
+
+  const marginTop = {
+    marginTop: theme.spacing(2),
+  };
+
+  return (
+    <>
+      {record && (
```

Comment:
This seems okay. An alternative would be to do
```typescript
return (
  {record
    ? <DialogBox>
      ...
      </DialogBox>
    : null}
);
```
Which would avoid returning an empty fragment. You'd have to change the return type to `JSX.Element | null` though.